packages/wonder-blocks-form/src/components/text-field.js

```diff
@@ -7,16 +7,48 @@ import Color from "@khanacademy/wonder-blocks-color";
 import Spacing from "@khanacademy/wonder-blocks-spacing";
 import {styles as typographyStyles} from "@khanacademy/wonder-blocks-typography";
 
-export default class TextField extends React.Component<{||}> {
+type State = {|
+    enabled: boolean,
+    disabled: boolean,
+    error: ?string,
+    focused: boolean,
+|};
+
+export default class TextField extends React.Component<{||}, State> {
+    state: State = {
+        enabled: true,
+        disabled: false,
+        error: null,
+        focused: false,
+    };
+
+    handleOnFocus: () => void = () => {
+        this.setState({
+            focused: true,
+        });
+    };
+
+    handleOnBlur: () => void = () => {
+        this.setState({
+            focused: false,
+        });
+    };
+
     render(): React.Node {
         return (
             <input
                 className={css([
                     styles.input,
                     typographyStyles.LabelMedium,
-                    styles.default,
+                    this.state.enabled && styles.default,
```

Comment:
Based on the first comment, this would be translated to `styles.default, ....` and it's replaced with whatever UI state changes in the component (via `state` with `error, focused` or via `props` with `disabled`).